dbms/src/Flash/Coprocessor/TiDBTableScan.cpp

```diff
@@ -24,6 +24,7 @@ TiDBTableScan::TiDBTableScan(
     , executor_id(executor_id_)
     , is_partition_table_scan(table_scan->tp() == tipb::TypePartitionTableScan)
     , columns(is_partition_table_scan ? table_scan->partition_table_scan().columns() : table_scan->tbl_scan().columns())
+    , keep_order(table_scan->tbl_scan().has_keep_order() ? table_scan->tbl_scan().keep_order() : true)
```

Comment:
If `is_partition_table_scan` is true, keep_order can be set to false since partition table scan will not require keep order.

Reply:
Does the table scan executor of partition table will set `keep_order` to be `false`?

Reply:
For patition table scan, you should get it by `table_scan->partition_table_scan()` and there is no `keep_order` field in `partition_table_scan`

Reply:
https://github.com/pingcap/tiflash/pull/5377/commits/93c92bf2a50df874beb6d66769e9db539bbda35c

@windtalker From better understanding, I have refined the code as below. PTAL.
```cpp
    if (is_partition_table_scan || (table_scan->tbl_scan().has_keep_order() && table_scan->tbl_scan().keep_order() == false))
    {
        keep_order = false;
    }
    else
    {
        keep_order = true;
    }
```